src/tls.c

```diff
@@ -3643,6 +3640,14 @@ static int TLSX_SupportedCurve_Parse(WOLFSSL* ssl, byte* input, word16 length,
     if (offset == length)
         return 0;
 
+    if(!isRequest && !IsAtLeastTLSv1_3(ssl->version)) {
```

Comment:
As discussed... can you move this block up to line 3630 to be more like previous behavior? Thanks